packages/interaction-kit/src/interactions/index.ts

```diff
@@ -217,16 +213,21 @@ export function handler(
 		}
 
 		case InteractionType.MessageComponent: {
-			const component = application.getComponent(json.data.custom_id);
+			let component = application.getComponent(json.data.custom_id);
+
+			if (component == null) {
+				component = await application.findComponent(json.data.custom_id);
```

Comment:
You could just put this on the if below, right? Since they are the same, I guess.

Reply:
Not quite - the one below will check to see if this also returns null, and if so will throw an error since it doesn't know how to route the component